contracts/gateway/AxelarAmplifierGateway.sol

```diff
@@ -8,24 +8,35 @@ import { IAxelarGatewayWeightedAuth } from '../interfaces/IAxelarGatewayWeighted
 import { ECDSA } from '../libs/ECDSA.sol';
 
 contract AxelarAmplifierGateway is IAxelarAmplifierGateway {
+    /// @dev This slot contains all the storage for this contract in an upgrade-compatible manner
     // keccak256('AxelarAmplifierGateway.Slot') - 1;
     bytes32 internal constant AXELAR_AMPLIFIER_GATEWAY_SLOT =
         0xca458dc12368669a3b8c292bc21c1b887ab1aa386fa3fcc1ed972afd74a330ca;
 
     struct AxelarAmplifierGatewayStorage {
+        string chainName;
+        string router;
         mapping(bytes32 => bool) commands;
         mapping(bytes32 => bool) approvals;
     }
 
-    bytes32 internal constant SELECTOR_APPROVE_CONTRACT_CALL = keccak256('approveContractCall');
-    bytes32 internal constant SELECTOR_TRANSFER_OPERATORSHIP = keccak256('transferOperatorship');
-
     IAxelarGatewayWeightedAuth public immutable authModule;
-
-    constructor(address authModule_) {
+    bytes32 public immutable chainNameHash;
+    bytes32 public immutable routerHash;
+
+    constructor(
+        address authModule_,
+        string memory chainName,
+        string memory router
+    ) {
         if (authModule_.code.length == 0) revert InvalidAuthModule();
 
         authModule = IAxelarGatewayWeightedAuth(authModule_);
+
+        _storage().chainName = chainName;
```

Comment:
Maybe cache the slot? And call it gatewayStorage so it's conflicting with any other private function for storage?

Reply:
it's a private method, so i'm thinking we use `_storage` as a convention for all contracts